### PR TITLE
Adding of Monarch keyboard

### PR DESCRIPTION
**Pull request by RinoSolutions**
## Description

Adding Monarch 65 keyboard to QMK.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/monarch/monarch.c [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "monarch.h"
```
Comment (drashna) [inline keyboards/monarch/monarch.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/monarch/monarch.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#pragma once
```
Comment (drashna) [inline keyboards/monarch/monarch.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/monarch/config.h [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+#define MATRIX_COL_PINS { A10, A9, A8, B15, B14, B13, B12, B10, B2, B1, B0, A5, A7, A4, A3, B6 }
+#define MATRIX_ROW_PINS { A15, B3, B11, A2, A1, B9 }
+#define DIODE_DIRECTION COL2ROW
+
+/* Rotary encoder pins */
+#define ENCODERS_PAD_A { B5 }
+#define ENCODERS_PAD_B { B4 }
+#define ENCODER_RESOLUTION 1
+#define ENCODER_DIRECTION_FLIP
```
Comment (zvecr) [inline keyboards/monarch/config.h:39]:
```suggestion
#define ENCODERS_PAD_A { B4 }
#define ENCODERS_PAD_B { B5 }
#define ENCODER_RESOLUTION 1
```

File: keyboards/monarch/keymaps/via/config.h [hunk 4]
```diff
@@ -0,0 +1 @@
+#define DYNAMIC_KEYMAP_EEPROM_MAX_ADDR 2047
```
Comment (zvecr) [inline keyboards/monarch/keymaps/via/config.h:1]:
```suggestion
#pragma once

#define DYNAMIC_KEYMAP_EEPROM_MAX_ADDR 2047
```

Also needs licence header

File: keyboards/monarch/keymaps/via/rules.mk [hunk 5]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
+BOOTMAGIC_ENABLE = lite
```
Comment (zvecr) [inline keyboards/monarch/keymaps/via/rules.mk:2]:
```suggestion
```

File: keyboards/monarch/readme.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+![Wete Keyboard](https://i.imgur.com/dZ4FRar.jpg)
+
+A southpaw full-sized keyboard.
+
+* Keyboard Maintainer: [Ramon Imbao](https://github.com/ramonimbao)
+* Hardware Supported: STM32F072CBT6
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ramonimbao/wete:default
```
Comment (zvecr) [inline keyboards/monarch/readme.md:12]:
```suggestion
    make monarch:default
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/monarch/monarch.c (hunk 1 above)
Comment (RinoSolutions) [reply keyboards/monarch/monarch.c:15]: I added the header I believe

File: keyboards/monarch/monarch.h (hunk 2 above)
Comment (RinoSolutions) [reply keyboards/monarch/monarch.h:15]: I believe I added the header

**Review by RinoSolutions — COMMENTED** [review]
I'm new to all this so hopefully I have done what you needed.
Thanks so much!

**Review by drashna — APPROVED** [review]

File: keyboards/monarch/keymaps/via/rules.mk (hunk 5 above)
Comment (RinoSolutions) [reply keyboards/monarch/keymaps/via/rules.mk]: Is this all set now?

File: keyboards/monarch/readme.md (hunk 6 above)
Comment (RinoSolutions) [reply keyboards/monarch/readme.md]: Have I made the proper adjustments?

File: keyboards/monarch/keymaps/via/config.h (hunk 4 above)
Comment (RinoSolutions) [reply keyboards/monarch/keymaps/via/config.h:18]: Made updates

File: keyboards/monarch/config.h (hunk 3 above)
Comment (RinoSolutions) [reply keyboards/monarch/config.h]: Am I good to go now? I think I’ve updated all necessary files if you get latest.

**Review by drashna — APPROVED** [review]

**Comment by RinoSolutions** [comment]
Good Morning,

I'm still waiting for someone to approve this request so it can be merged into the main repository.
I believe everything is completed that needs to be.
Let me know what I need to do if there is something else I have to do.
My keyboard is being shipped to buyers soon and I'd love to have this available in the main repo.

Thanks!
________________________________

**Comment by RinoSolutions** [comment]
Hello there,
I'm still waiting for someone to approve this. 
My keyboard is going to be shipping to buyers soon and I would love to have this merged into the main repo.
Let me know if there is something else I need to do in order to get this completed.
Thanks!

File: keyboards/monarch/rules.mk [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no # Custom matrix file
+BACKLIGHT_ENABLE = yes
+RGBLIGHT_ENABLE = no
+ENCODER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/monarch/rules.mk:18]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

File: keyboards/monarch/readme.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
+# Monarch
+
+![Monarch Keyboard](https://i.imgur.com/3nnybLk.jpg)
```
Comment (fauxpark) [inline keyboards/monarch/readme.md:3]:
```suggestion
![Monarch Keyboard](https://i.imgur.com/3nnybLkl.jpg)
```

File: keyboards/monarch/readme.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+A 65percent keyboard with a rotary encoder and interchangeable inserts.
+www.kingkbs.com
+
+* Keyboard Maintainer: [Ramon Imbao](https://github.com/ramonimbao)
+* Hardware Supported: STM32F072CBT6
```
Comment (fauxpark) [inline keyboards/monarch/readme.md:9]:
```suggestion

* Keyboard Maintainer: [Ramon Imbao](https://github.com/ramonimbao)
* Hardware Supported: STM32F072CBT6
* Hardware Availability: https://www.kingkbs.com
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
The Chibios *conf.h files should be run through `qmk chibios-confmigrate`.

**Comment by RinoSolutions** [comment]
Thank you for the quick review. 

I don't know how to run the qmk chibios-confmigrate 

I tried executing it but its telling me that qmk doesn't have a chibios-confmigrate argument.

I'm sorry for the request for help but I did not design the pcb and the designer has left the project so I'm doing my best to work through these issues. 

Any help you can offer is greatly appreciated.

Thanks!

**Comment by fauxpark** [comment]
Sounds like your repo is out of date, you will need to rebase this branch onto qmk:master and run `python3 -m pip install -U qmk` to update the CLI as well.

**Review by fauxpark — COMMENTED** [review]
You can probably simply delete those three files, in fact. The stuff that is different to the ones under `platforms/chibios/` is probably not relevant (eg. I2C is enabled, but you don't have anything enabled like OLEDs which would use I2C).

**Comment by RinoSolutions** [comment]
I'm sorry but I don't know what 3 files you're referencing. 
I pulled down the patch-1 branch again, checked it out, ran the python3 line you have above, but I can't get the rebase to work.
I ran `git rebase -i origin/master` and it looks like it entered into some module where it showed me changes but I wasn't able to do anything in there like pick/squash/etc
Again I'm very sorry for the hassle. The pcb designer was supposed to take care of all this and left me struggling to work through it.

**Comment by RinoSolutions** [comment]
Also, when I run `qmk chibios-confmigrate -i keyboards/monarch/chconf.h` 
I receive the following....

<class 'TypeError'>
ERROR expected str, bytes or os.PathLike object, not NoneType
Traceback (most recent call last):
  File "C:/msys64/mingw64/lib/python3.8/site-packages/milc/milc.py", line 467, in __call__
    return self.__call__()
  File "C:/msys64/mingw64/lib/python3.8/site-packages/milc/milc.py", line 472, in __call__
    return self._entrypoint(self)
  File "C:/msys64/home/rnort/qmk_firmware/lib/python/qmk/cli/chibios/confmigrate.py", line 120, in chibios_confmigrate
    reference_defs = collect_defines(cli.args.reference)
  File "C:/msys64/home/rnort/qmk_firmware/lib/python/qmk/cli/chibios/confmigrate.py", line 43, in collect_defines
    with open(filepath, 'r') as f:
TypeError: expected str, bytes or os.PathLike object, not NoneType

**Comment by fauxpark** [comment]
The three files are the *conf.h files. As for the error, I think you are missing the `-r` argument. For the chconf and halconf, the reference files are in `platforms/chibios/common/configs`, and the mcuconf will be the one under `GENERIC_STM32_F072XB/configs`.

**Review by RinoSolutions — COMMENTED** [review]
I believe I've updated the 3 *conf.h files properly. 
If you see any issues please let me know and I'll do my best to correct them.
Once again thank you for the assistance.

**Review by fauxpark — CHANGES_REQUESTED** [review]
The *conf.h files can be deleted.

**Review by RinoSolutions — COMMENTED** [review]
No problem. I deleted the files as requested.

I see there is another pending review as well, is there something I need to do for that one?

Thanks again very much!

**Review by fauxpark — APPROVED** [review]

**Review by RinoSolutions — COMMENTED** [review]
Hi there. 
I believe I've made all the changes as you've requested. 
Please let me know if there's anything else I need to do.
Some of the changes have been awaiting review for a couple months so I didn't know if there is something I need to trigger, or just wait.
Thank you again.

**Comment by drashna** [comment]
Thanks!

**Pull request merged**